gems/laa_multi_step_forms/docs/DSLDecisionTree.md

```diff
@@ -0,0 +1,136 @@
+
+### Simple transition from A -> B
+
+```
+from('A').goto(show: 'B')
+from('B').goto(edit: 'C')
+from('C').goto(index: 'D')
+
+it_behaves_like 'a generic decision', from: 'A', goto: { action: :show, controller: 'B' }
+it_behaves_like 'a generic decision', from: 'B', goto: { action: :edit, controller: 'C' }
+it_behaves_like 'a generic decision', from: 'C', goto: { action: :index, controller: 'D' }
+```
+
+The goto must have a key of `show`, `edit` or `index`, which is used to set the action and
+controller of the redirect.
+
+### Transition with a conditional
+
+The block in the `when` condition is applied to the form object that is passed in, so in this instance it is
+expecting the form object to expose a `add_another` object/method.
+
+```
+from('A)
+  .when(-> { add_another.yes? })
+  .goto(edit: 'B')
+  .goto(edit: 'C')
+
+context 'answer yes to add_another' do
+  before { allow(form).to receive(:add_another).and_return(YesNoAnswer::YES) }
+
+  it_behaves_like 'a generic decision', from: 'A', goto: { action: :edit, controller: 'B' }
+end
+
+context 'answer no to add_another' do
+  before { allow(form).to receive(:add_another).and_return(YesNoAnswer::NO) }
+
+  it_behaves_like 'a generic decision', from: 'A', goto: { action: :edit, controller: 'C' }
+end
+```
+
+### Additional parameters in the goto location
+
+Additional parameters can be set in 3 ways:
+1. static value -> this just adds the addiitonal paramas in the hash
+2. dynamic value (block without params) -> this executes on the block form object allowing the value to be set
+3. dynamic value (block with params) -> this passes the result of the `when` block into the `goto` block
+
+> NOTE: type 3 is not currently in use and unsure of a required use-case - documented for completeness
+
+```
+from('A').goto(edit: 'B', tag: 'help')
+from('B').goto(edit: 'C', tag: -> { record.id })
+from('C')
+  .when(-> { record }).goto(edit: 'D', tag: ->(inst) { inst.id })
+
+it_behaves_like 'a generic decision', from: 'A', goto: { action: :edit, controller: 'B', tag: 'help' }
+context do
+  let(:tag) { record.id }
+  it_behaves_like 'a generic decision', from: 'B', goto: { action: :edit, controller: 'C', tag: 'help' }, additional_param: :tag
+end
+context do
+  let(:tag) { record.id }
+  it_behaves_like 'a generic decision', from: 'C', goto: { action: :edit, controller: 'D', tag: 'help' }, additional_param: :tag
+end
+```
+
+> NOTE: for non-static values the tests expects a `let` to expose the tag value (must have matching name to param) to exist that
+> can be called from the text scope.
+
+### Additional logic functionality
+
+Sometime is may be helpful to have addition or conplex logic that is used in multiple `when` checks into the
+tree. This is possible by overwriting the `WRAPPER_CLASS` constant in the custom DesicionTree class. This is
+then used to wrap the form object and allow additional feature to be shared without populting the form object
+or desicion tree with the logic.
```

Comment:
```suggestion
Sometimes it may be helpful to have additional or complex logic that is used in multiple `when` checks in the
tree. This is possible by overwriting the `WRAPPER_CLASS` constant in the custom DecisionTree class. This is
then used to wrap the form object and allow additional features to be shared without polluting the form object
or decision tree with the logic.
```